pandas/core/arrays/arrow/array.py

```diff
@@ -1634,6 +1634,10 @@ def _replace_with_mask(
                 indices = pa.array(indices, type=pa.int64())
                 replacements = replacements.take(indices)
             return cls._if_else(mask, replacements, values)
+        if isinstance(values, pa.ChunkedArray):
```

Comment:
You could limit it to combine the chunks only when the values have boolean type

Reply:
done